modules/chapter1/pages/section1.adoc

```diff
@@ -18,27 +18,38 @@ Good machine learning scenarios often have the following common properties:
 
+== Choosing your algorithm
+We have a plethora of mathematical algorithms that may, or may not, solve our problem and be applied on our data. But, how do you recognise the category of problem and the algorithm that can help you solve it?
+The clue lies with your outcome variable.
+ 
+In a classification problem, the outcome variable (the event you want to predict) is a category or class label. It can be binary, such as "Sick" or "Healthy", "Yes" or "No". They can be multi-class such as for image recognition when you want to identify more than 2 classes or labels. 
+
+In a regression problem, the outcome variable is a continuous value and the goal is predicting a quantitative value. Common examples are predicting house prices, temperatures and sale revenues. 
+
+Knowing the outcome variable you are able to identify if its supervised, unsupervised, semi-supervised or reinforcement learning. However, there is a lot of work left before the final algorithm is chosen. The data scientist will spend time exploring the data set, applying different visualization and data cleaning technqiues exploring the underlying data, and several different machine learning algorithms may be applied to test which one solves the problem. Solving the problem is multifacetted and there is no one algorithm to rule them all.  
+ 
 == What is a Machine Learning model
 
-A Machine Learning model is a file that has been trained to recognize certain types of patterns. You train a model over a set of data, providing it with an algorithm that it can use to reason over and learn from that data.
+A Machine Learning model is a file that has been trained to recognize certain types of patterns. You train a mdel over a set of data, providing it with an algorithm that it can use to reason over and learn from that data.
 
-Once you have trained the model, you can use it to reason over data that it has not seen before, and make predictions about that data. For example, let's say you want to build an application that can recognize a user's emotions based on their facial expressions. You can train a model by providing it with images of faces that are each tagged with a certain emotion, and then you can use that model in an application that can recognize any user's emotion.
+Once you have trained the model, you can use it to reason over data that it has not seen before but comes from the same applicability domain, and make predictions about that data. For example, let's say you want to build an application that can recognize a user's emotions based on their facial expressions. You can train a model by providing it with images of faces that are each tagged with a certain emotion, and then you can use that model in an application that can recognize any user's emotion.
```

Comment:
What if we briefly mention the "generalization" concept?

```suggestion
Once you have trained the model, you can use it to reason over data that it has not seen before but comes from the same applicability domain, and make predictions about that data. This ability, called _generalization_, is a key concept in machine learning, as it refers to the capacity of models to adapt to unseen data.

 For example, let's say you want to build an application that can recognize a user's emotions based on their facial expressions. You can train a model by providing it with images of faces that are each tagged with a certain emotion, and then you can use that model in an application that can recognize any user's emotion.
```

Reply:
Love it!